apps/shared/avifpng.c

```diff
@@ -119,46 +95,95 @@ static avifBool avifCopyRawProfile(const char * profile, size_t profileLength, a
 {
 #ifdef PNG_eXIf_SUPPORTED
-    png_uint_32 exifSize = 0;
-    png_bytep exif = NULL;
-    if (png_get_eXIf_1(png, info, &exifSize, &exif) == PNG_INFO_eXIf) {
-        if ((exifSize == 0) || !exif) {
-            fprintf(stderr, "Exif extraction failed: empty eXIf chunk\n");
-            return AVIF_FALSE;
+    if (!*ignoreExif) {
+        png_uint_32 exifSize = 0;
+        png_bytep exif = NULL;
+        if (png_get_eXIf_1(png, info, &exifSize, &exif) == PNG_INFO_eXIf) {
+            if ((exifSize == 0) || !exif) {
+                fprintf(stderr, "Exif extraction failed: empty eXIf chunk\n");
+                return AVIF_FALSE;
+            }
+            avifImageSetMetadataExif(avif, exif, exifSize);
+            *ignoreExif = AVIF_TRUE; // Ignore any other Exif chunk.
         }
-        avifImageSetMetadataExif(avif, exif, exifSize);
-        return AVIF_TRUE;
     }
 #endif // PNG_eXIf_SUPPORTED
 
+    // HEIF specification ISO-23008 section A.2.1 allows including and excluding the Exif\0\0 header from AVIF files.
+    // The PNG 1.5 extension mentions the omission of this header for the modern standard eXIf chunk.
+    const avifROData exifApp1Header = { (const uint8_t *)"Exif\0\0", 6 };
+    const avifROData xmpApp1Header = { (const uint8_t *)"http://ns.adobe.com/xap/1.0/\0", 29 };
+
+    // tXMP could be retrieved using the png_get_unknown_chunks() API but tXMP is deprecated
+    // and there is no PNG file example with a tXMP chunk lying around, so it is not worth the hassle.
+
     png_textp text = NULL;
     const png_uint_32 numTextChunks = png_get_text(png, info, &text, NULL);
-    for (png_uint_32 i = 0; i < numTextChunks; ++i, ++text) {
-        if (!strcmp(text->key, "Raw profile type exif") || !strcmp(text->key, "Raw profile type APP1")) {
-            png_size_t textLength;
-            switch (text->compression) {
+    for (png_uint_32 i = 0; (!*ignoreExif || !*ignoreXMP) && (i < numTextChunks); ++i, ++text) {
+        png_size_t textLength = text->text_length;
 #ifdef PNG_iTXt_SUPPORTED
-                case PNG_ITXT_COMPRESSION_NONE:
-                case PNG_ITXT_COMPRESSION_zTXt:
-                    textLength = text->itxt_length;
-                    break;
+        if ((text->compression == PNG_ITXT_COMPRESSION_NONE) || (text->compression == PNG_ITXT_COMPRESSION_zTXt)) {
+            textLength = text->itxt_length;
+        }
 #endif
-                case PNG_TEXT_COMPRESSION_NONE:
-                case PNG_TEXT_COMPRESSION_zTXt:
-                default:
-                    textLength = text->text_length;
-                    break;
+
+        if (!*ignoreExif && !strcmp(text->key, "Raw profile type exif")) {
+            if (!avifCopyRawProfile(text->text, textLength, &avif->exif)) {
+                return AVIF_FALSE;
+            }
+            avifRemoveHeader(&exifApp1Header, &avif->exif); // Optional.
```

Comment:
Nit: It's not clear what the comment "Optional" means. I think it means we don't need to check the return value of `avifRemoveHeader()` because the header is optional. If so, it would be good to say that, e.g., "Ignore the return value because the header is optional."